scrape/scrape.go

```diff
@@ -967,12 +967,13 @@ mainLoop:
 
 		// A failed scrape is the same as an empty scrape,
 		// we still call sl.append to trigger stale markers.
-		total, added, seriesAdded, appErr := sl.append(b, contentType, start)
+		scrapeAppender := sl.appender()
+		total, added, seriesAdded, appErr := sl.append(scrapeAppender, b, contentType, start)
 		if appErr != nil {
 			level.Debug(sl.l).Log("msg", "append failed", "err", appErr)
 			// The append failed, probably due to a parse error or sample limit.
 			// Call sl.append again with an empty scrape to trigger stale markers.
-			if _, _, _, err := sl.append([]byte{}, "", start); err != nil {
+			if _, _, _, err := sl.append(scrapeAppender, []byte{}, "", start); err != nil {
```

Comment:
Don't we need a new appender here?

Reply:
thanks .done

Reply:
We still need a new appender here, and the old one rolled back.